webpack.common.js

```diff
@@ -0,0 +1,58 @@
+const path = require('path')
+const webpack = require('webpack')
+const HtmlWebpackPlugin = require('html-webpack-plugin')
+const InterpolateHtmlPlugin = require('interpolate-html-plugin')
+
+const publicUrl = ''
+
+module.exports = {
+    entry: {
+        app: './src/index.tsx'
+    },
+    output: {
+        filename: 'bundle.js',
+        path: path.join(__dirname, 'dist'),
+        publicPath: `${publicUrl}/`
+    },
+
+    resolve: {
+        // Add '.ts' and '.tsx' as resolvable extensions.
+        extensions: ['.ts', '.tsx', '.js', '.json'],
+        modules: [path.resolve(__dirname, 'src'), path.resolve(__dirname, 'node_modules')]
+    },
+
+    module: {
+        rules: [
+            // All files with a '.ts' or '.tsx' extension will be handled by 'awesome-typescript-loader'.
+            {
+                test: /\.(ts|tsx)?$/,
+                loader: ['awesome-typescript-loader'],
+                exclude: /node_modules/
+            },
+            {
+                test: /\.css$/,
+                use: [
+                    {
+                        loader: 'style-loader'
+                    },
+                    {
+                        loader: 'css-loader'
+                    }
+                ]
+            },
+            // All output '.js' files will have any sourcemaps re-processed by 'source-map-loader'.
+            { enforce: 'pre', test: /\.(ts|tsx)$/, loader: 'source-map-loader' }
+        ]
+    },
+    plugins: [
+        new HtmlWebpackPlugin({
+            template: './public/index.html'
+        }),
+        new InterpolateHtmlPlugin({
+            PUBLIC_URL: publicUrl
+        }),
+        new webpack.HotModuleReplacementPlugin()
```

Comment:
maybe move this plugin to dev config [https://webpack.js.org/guides/hot-module-replacement/](url)